Plugins/Public/cloak_plugin/Cloak.cpp

```diff
@@ -291,51 +309,40 @@ void SetState(uint iClientID, uint iShipID, int iNewState)
 	}
 }
 
-bool removeSingleFuel(uint iClientID, CLOAK_INFO &info)
-{
-	for (list<EquipDesc>::iterator item = Players[iClientID].equipDescList.equip.begin(); item != Players[iClientID].equipDescList.equip.end(); item++)
-	{
-		uint fuel_usage = info.arch.mapFuelToUsage[item->iArchID];
-		if (item->iCount >= fuel_usage)
-		{
-			pub::Player::RemoveCargo(iClientID, item->sID, fuel_usage);
-			info.singleCloakConsumed = true;
-			return true;
-		}
-	}
-
-	return false;
-
-}
-
 // Returns false if the ship has no fuel to operate its cloaking device.
-static bool ProcessFuel(uint iClientID, CLOAK_INFO &info)
+static bool ProcessFuel(uint iClientID, CLOAK_INFO &info, uint iShipID)
 {
 	if (info.bAdmin)
 		return true;
 
-	if (!info.singleCloakConsumed && info.arch.singleUseCloak) {
-		return removeSingleFuel(iClientID, info);
-	}
-
-	else if (info.singleCloakConsumed)
-	{
+	if(setJumpingClients.find(iClientID) != setJumpingClients.end())

```

Comment:
why are we adding logic around jumping while cloaking here? if you're jumping then you don't need any cloaking fuel removed?

Reply:
Should probably be renamed. When in a jump tunnel, player ship is moving at an insane speed, around 10kps. This was added not to instantly drain their fuel the moment they hit a jumphole/gate

Reply:
Oh that's interesting to hear written down and stated as fact. I always had a suspicion that while in the jump tunnel the player is actually still in one of the systems somewhere moving at speed, but never proved it. Makes sense.